subprojects/common/math/src/main/kotlin/com/avito/math/Percent.kt

```diff
@@ -23,32 +25,44 @@ public fun Double.fromZeroToOnePercent(): Percent {
 @Suppress("MagicNumber")
 public fun Float.fromZeroToOnePercent(): Percent {
     require(this in 0.0..1.0) { "Trying to convert $this to percents; should be in [0.0..1.0] range" }
-    return NumberPercent(this * 100)
+    return BigDecimalTwoDigitsPrecisionPercent.create(this * 100)
 }
 
 public fun Int.fromZeroToHundredPercent(): Percent {
     require(this in 0..100) { "Trying to convert $this to percents; should be in [0..100] range" }
-    return NumberPercent(this)
+    return BigDecimalTwoDigitsPrecisionPercent.create(this)
 }
 
 public interface Percent {
 
-    public fun twoDigitsString(): String
+    public fun roundToTwoDigitsPrecision(): String
 
-    public fun toInt(): Int
+    public fun roundToInt(): Int
 
-    public fun toLong(): Long
+    public fun roundToLong(): Long
 }
 
-internal data class NumberPercent(val value: Number) : Percent {
+internal data class BigDecimalTwoDigitsPrecisionPercent private constructor(val value: BigDecimal) : Percent {
```

Comment:
Proposal for renaming:
`BigDecimalTwoDigitsPrecisionPercent` --> `TwoDigitsPrecisionPercent|TwoDigitsPercent`

Because from the client perspective BigDecimal is implementation details that are already hidden by API.

Reply:
To be honest, the correct name will be FourDigitsPercent